File: 백준/Silver/18110. solved．ac/solved．ac.py
import sys

n = int(sys.stdin.readline())
if n == 0:
    sys.stdout.write("0\n")
else:
    arr = []
    
    for _ in range(n):
        arr.append(int(sys.stdin.readline()))
    
    arr = sorted(arr)
    cnt = int(n*0.15 + 0.5)

    sys.stdout.write(f"{int(sum(arr[cnt:n-cnt]) / len(arr[cnt:n-cnt]) + 0.5)}\n")

File: 백준/Silver/18110. solved．ac/solved．ac.cc
#include <iostream>
#include <vector>
#include <cmath>
#include <algorithm>
using namespace std;

int main() {
    ios::sync_with_stdio(false); cin.tie(NULL); cout.tie(NULL);

    int n;
    cin >> n;

    if (n == 0)
        cout << 0 << "\n";
    else {
        vector<int> arr(n);

        for (int i = 0; i < n; i++)
            cin >> arr[i];
    
        sort(arr.begin(), arr.end());
        
        int cnt = round(n * 0.15);
        double sum = 0;

        for (int i = cnt; i < n-cnt; i++)
            sum += arr[i];

        cout << round(sum / (n - cnt*2)) << "\n";
    }

    return 0;
}